### ISSUE TITLE
Add a "capital_gain" value expression variable

### ISSUE THREAD

**Issue by blatte** [ledger/ledger#1996]
Just like there's the "amount" value expression variable, it would be useful to have something like "capital_gain" value expression variable for the realized capital gain (or loss) on the account in the posting.

Since this is possible with the "-G" option in reports, maybe this could be added. One of the obvious use case would be to automatically calculate taxes on profits made on selling stocks.

**Comment by Christoph-D** [comment]
Ledger's double-accounting method already tracks capital gains in each posting explicitly or the amounts wouldn't match (example on https://www.ledger-cli.org/3.0/doc/ledger3.html#Buying-and-Selling-Stock). The balance on this artificial `Income:Capital Gains` account shows the net gain/loss (the name is of course arbitrary).

The `-G` option works on the database of historical prices. I may be wrong but I don't think ledger uses this price database for balancing postings, only for reports to display plausible values of commodity accounts.

Depending on a countries tax laws, capital gains also exclude broker fees or other fees, which a `capital_gain` value expression would not be able to model. An account like `Income:Capital Gains` can model the gains correctly in these cases (it must, due to double-accounting). The example in the documentation above subtracts broker fees before putting the remaining difference into capital gains.

Could you give an example of how you'd like to use a `capital_gain` value expression and how it would be different from a capital gains account?

**Comment by blatte** [comment]
You are correct, the -G option is based on historical value and this is exactly what I wish for in a capital gain expression. Let's take the example from the doc.

Buying:
2004/05/01 Stock purchase
    Assets:Broker                     50 AAPL @ $30.00
    Expenses:Broker:Commissions        $19.95
    Assets:Broker                  $-1,519.95

Selling:
2005/08/01 Stock sale
    Assets:Broker                    -50 AAPL {$30.00} @ $50.00
    Expenses:Broker:Commissions        $19.95
    Income:Capital Gains           $-1,000.00
    Assets:Broker                   $2,480.05

I don't mind calculating the gain myself, but what if there are more transactions, buying and selling, over time like so: 

Buying at:
    Assets:Broker                     50 AAPL @ $30.00
Then at
    Assets:Broker                     20 AAPL @ $35.00

Then taking some profit:
    Assets:Broker                    -30 AAPL @ $45.00

And so on....

I would have to calculate gains by hand for different buying prices. The idea would be to have an automatic transaction like this for gains and taxes on those gains being processed:
=Assets:Broker
   Income:Capital Gains        (-(capital_gain-(capital_gain/1.3)))
   Expenses:Taxes     (capital_gain-(capital_gain/1.3))

This would need refinement of course but hopefully you get the general idea. And yes it is based on the capital_gain expression value being calculated according to historical value at the date of the transaction.

**Comment by Christoph-D** [comment]
So what you want is a posting like this:

```
2005/08/01 Stock sale
    Assets:Broker                    -50 AAPL {$30.00} @ $45.00
    Assets:Broker                    -20 AAPL {$35.00} @ $45.00
    Expenses:Broker:Commissions        $19.95
    Income:Capital Gains
    Assets:Broker                   $3130.05
```

but without having to calculate `$3130.05` by hand?

I think the posting needs to mention the `Assets:Broker` and `Income:Capital Gains` accounts in any case or it wouldn't balance, so `Income:Capital Gains` can't be only in an automatic transaction. But maybe it's possible to automatically calculate the amounts for both (I don't know).
